internal/common/node_module_info.bzl

```diff
@@ -12,63 +12,40 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""NodeModuleInfo & NodeModuleSources providers and apsect to collect node_modules from deps.
+"""NodeModuleInfo providers and apsect to collect node_modules from deps.
 """
 
-# NodeModuleInfo provider is only provided by targets that are npm dependencies by the
-# `node_module_library` rule. This provider is currently used by different rules to filter out
-# npm dependencies such as
-# ```
-# [d for d in ctx.attr.deps if not NodeModuleInfo in d]
-# ```
-# in `packages/typescript/internal/build_defs.bzl` or
-# ```
-# hasattr(target, "files") and not NodeModuleInfo in target:
-# ```
-# in `internal/common/sources_aspect.bzl`.
-# Similar filtering is done in downstream repositories such as angular/angular so this provider
-# needs to go through a deprecation period before it can be phased out.
-NodeModuleInfo = provider(
-    doc = "Provides information about npm dependencies installed with yarn_install and npm_install rules",
-    fields = {
-        "workspace": "The workspace name that the npm dependencies are provided from",
-    },
-)
-
-# NodeModuleSources provider is provided by targets that are npm dependencies by the
+# NodeModuleInfo provider is provided by targets that are npm dependencies by the
 # `node_module_library` rule as well as other targets that have direct or transitive deps on
-# `node_module_library` targets via the `collect_node_modules_aspect` below.
-# TODO: rename to NodeModuleSourcesInfo so name doesn't trigger name-conventions warning
-# buildozer: disable=name-conventions
-NodeModuleSources = provider(
-    doc = "Provides sources for npm dependencies installed with yarn_install and npm_install rules",
+# `node_module_library` targets via the `node_modules_aspect` below.
+NodeModuleInfo = provider(
+    doc = "Provides information about npm dependencies",
```

Comment:
FYI to remove js_library I think I'd like to produce this provider from npm_package too. Would be nice to include package name (could add in later PR)

Reply:
I had `sources` as direct and `transitive_sources` as direct + transitive but changed to just sources which are direct + transitive similar to what we decided for the JS providers. Does that sounds like the right approach for this?

+1 for adding `package` (or `module_name`) or later PR.

Reply:
I'm not sure. In JS tooling we sometimes want a strict-deps policy that requires distinguishing direct and transitive sources. I would bet we'll need direct for something sometime

Reply:
Hmmmm. I like the idea of being able to differentiate as well even in the new JS providers. It would have solved my karma runtime_deps issue I saw earlier today by being able to just access `sources` instead of having to take all of `transitive_sources`.